lib/TPP/Transforms/LinalgVectorize.cpp

```diff
@@ -0,0 +1,86 @@
+//===- LinalgVectorize.cpp ---------------------------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Passes.h"
+
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/Dialect/Tensor/Transforms/Transforms.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+#include "mlir/Dialect/Vector/Transforms/LoweringPatterns.h"
+#include "mlir/Dialect/Vector/Transforms/VectorRewritePatterns.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/IRMapping.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+
+namespace mlir {
+namespace tpp {
+#define GEN_PASS_DEF_LINALGVECTORIZE
+#include "TPP/Passes.h.inc"
+} // namespace tpp
+} // namespace mlir
+
+namespace {
+
+struct VectorizationPattern : public RewritePattern {
+  explicit VectorizationPattern(MLIRContext *context)
+      : RewritePattern(MatchAnyOpTypeTag(), /*benefit=*/1, context) {}
+  LogicalResult matchAndRewrite(Operation *op,
+                                PatternRewriter &rewriter) const override {
+    if (!linalg::hasVectorizationImpl(op))
+      return rewriter.notifyMatchFailure(op,
+                                         "Unsupported Op, cannot vectorize");
+    // Direct naive pack/unpack vectorization has terrible performance.
+    // Disable these for now and rely on other lowering patterns.
+    if (isa<linalg::PackOp, linalg::UnPackOp>(op))
+      return rewriter.notifyMatchFailure(op, "Packing vectorization disabled");
+    // Insert slice is vectorized into a vector read-write pair which by default
+    // introduces unnecessary extra operations.
+    // Disable insert vectorization for now and allow bufferization to fold it
+    // into subview in many cases.
+    if (isa<tensor::InsertSliceOp>(op))
+      return rewriter.notifyMatchFailure(op,
+                                         "Insert slice vectorization disabled");
+    return linalg::vectorize(rewriter, op);
+  }
+};
+
+// Based on 'vectorize_children_and_apply_patterns' transform op.
```

Comment:
Looks like this pass is a candidate for what Renato has asked for a couple times: passes that just wrap around a small schedule. Might be worth discussing if it is straightforward in this case or not.